adam/adam_6224.py

```diff
@@ -0,0 +1,628 @@
+        dtype='str',
+        access=AttrWriteType.READ,
+    )
+
+    Status_2 = attribute(
+        dtype='str',
+        access=AttrWriteType.READ,
+    )
+
+    Status_3 = attribute(
+        dtype='str',
+        access=AttrWriteType.READ,
+    )
+
+    EventStatus_0 = attribute(
+        dtype='str',
+        access=AttrWriteType.READ,
+    )
+
+    EventStatus_1 = attribute(
+        dtype='str',
+        access=AttrWriteType.READ,
+    )
+
+    EventStatus_2 = attribute(
+        dtype='str',
+        access=AttrWriteType.READ,
+    )
+
+    EventStatus_3 = attribute(
+        dtype='str',
+        access=AttrWriteType.READ,
+    )
+
+    # ---------------
+    # General methods
+    # ---------------
+
+    def init_device(self):
+        Device.init_device(self)
+        # PROTECTED REGION ID(ADAM6224.init_device) ENABLED START #
+        self.set_state(DevState.ON)
+        self.set_status("ADAM-6224 enabled")
+        # PROTECTED REGION END #    //  ADAM6224.init_device
+
+    def always_executed_hook(self):
+        # PROTECTED REGION ID(ADAM6224.always_executed_hook) ENABLED START #
+        # if not self.connected_ADAM.connect():
+        #     self.set_state(DevState.ALARM)
+        #     self.set_status("Connection  error")
+        pass
+        # PROTECTED REGION END #    //  ADAM6224.always_executed_hook
+
+    def delete_device(self):
+        # PROTECTED REGION ID(ADAM6224.delete_device) ENABLED START #
+        self.connected_ADAM.close()
+        # PROTECTED REGION END #    //  ADAM6224.delete_device
+
+    # ------------------
+    # Attributes methods
+    # ------------------
+
+    # --------------------
+    # DigitalInput methods
+    # --------------------
+
+    def read_DigitalInput_0(self):
+        # PROTECTED REGION ID(ADAM6224.DigitalInput_0_read) ENABLED START #
+        tmp = self.connected_ADAM.read_coils(0, 4)
+        self.digital_input_values = tmp.bits
+        return self.digital_input_values[0]
+        # PROTECTED REGION END #    //  ADAM6224.DigitalInput_0_read
+
+    def read_DigitalInput_1(self):
+        # PROTECTED REGION ID(ADAM6224.DigitalInput_1_read) ENABLED START #
+        return self.digital_input_values[1]
+        # PROTECTED REGION END #    //  ADAM6224.DigitalInput_1_read
+
+    def read_DigitalInput_2(self):
+        # PROTECTED REGION ID(ADAM6224.DigitalInput_2_read) ENABLED START #
+        return self.digital_input_values[2]
+        # PROTECTED REGION END #    //  ADAM6224.DigitalInput_2_read
+
+    def read_DigitalInput_3(self):
+        # PROTECTED REGION ID(ADAM6224.DigitalInput_3_read) ENABLED START #
+        return self.digital_input_values[3]
+        # PROTECTED REGION END #    //  ADAM6224.DigitalInput_3_read
+
+    # --------------------
+    # EventStatus methods
+    # --------------------
+
+    def read_EventStatus_0(self):
+        # PROTECTED REGION ID(ADAM6224.EventStatus_0_read) ENABLED START #
+        tmp = self.connected_ADAM.read_holding_registers(110, 4)
```

Comment:
Same situation here. The reading from the device is done only for the `EventStatus_0`. That doesn't look correct.